compiler/luci-interpreter/src/kernels/Add.test.cpp

```diff
@@ -27,57 +27,106 @@ namespace
+           std::initializer_list<int32_t> output_shape, std::initializer_list<float> input1_data,
+           std::initializer_list<float> input2_data, std::initializer_list<float> output_data)
+{
+  float min, max;
+  min =
+      std::min(input1_data) > std::min(input2_data) ? std::min(input2_data) : std::min(input1_data);
+  max =
+      std::max(input1_data) > std::max(input2_data) ? std::max(input1_data) : std::max(input2_data);
+  float kQuantizedTolerance = GetTolerance(min, max);
+  std::pair<float, int32_t> input_quant_param = quantizationParams<T>(min, max);
+  Tensor input1_tensor{getElementType<T>(),
+                       input1_shape,
+                       {{input_quant_param.first}, {input_quant_param.second}},
+                       ""};
+  Tensor input2_tensor{getElementType<T>(),
+                       input2_shape,
+                       {{input_quant_param.first}, {input_quant_param.second}},
+                       ""};
+  if (std::is_floating_point<T>::value)
   {
-    Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>(base_shape, input1_data);
-    Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>(test_shapes[i], input2_data);
-    Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+    input1_tensor.writeData(input1_data.begin(), input1_data.size() * sizeof(T));
+    input2_tensor.writeData(input2_data.begin(), input2_data.size() * sizeof(T));
+  }
+  else
+  {
+    std::vector<T> quantized_input1_value =
+        quantize<T>(input1_data, input_quant_param.first, input_quant_param.second);
+    std::vector<T> quantized_input2_value =
+        quantize<T>(input2_data, input_quant_param.first, input_quant_param.second);
+    input1_tensor.writeData(quantized_input1_value.data(),
+                            quantized_input1_value.size() * sizeof(T));
+    input2_tensor.writeData(quantized_input2_value.data(),
+                            quantized_input2_value.size() * sizeof(T));
+  }
+  std::pair<float, int32_t> output_quant_param = quantizationParams<T>(0, max * 2);
+  Tensor output_tensor =
+      makeOutputTensor(getElementType<T>(), output_quant_param.first, output_quant_param.second);
 
-    AddParams params{};
-    params.activation = Activation::RELU;
+  AddParams params{};
+  params.activation = Activation::RELU;
 
-    Add kernel(&input1_tensor, &input2_tensor, &output_tensor, params);
-    kernel.configure();
-    kernel.execute();
+  Add kernel(&input1_tensor, &input2_tensor, &output_tensor, params);
+  kernel.configure();
+  kernel.execute();
 
-    EXPECT_THAT(extractTensorData<float>(output_tensor),
-                ::testing::ElementsAreArray(ArrayFloatNear(test_outputs[i], 0.0001f)))
-        << "With shape number " << i;
+  if (std::is_floating_point<T>::value)
+  {
+    EXPECT_THAT(extractTensorData<T>(output_tensor),
+                ElementsAreArray(ArrayFloatNear(output_data, kQuantizedTolerance)));
```

Comment:
This is floating point test. Why do you use kQuantizedTolerance?

Reply:
Okay! I will update it!